src/views/Residents.js

```diff
@@ -0,0 +1,176 @@
+import React, { useState, useEffect, useCallback, useMemo } from 'react';
+import { Grid, H4, HelpText } from 'indigo-react';
+import { Just, Nothing } from 'folktale/maybe';
+import * as ob from 'urbit-ob';
+
+import View from 'components/View';
+import Tabs from 'components/Tabs';
+import Crumbs from 'components/Crumbs';
+import Sigil from 'components/Sigil';
+import Blinky from 'components/Blinky';
+import NavHeader from 'components/NavHeader';
+
+import { useLocalRouter } from 'lib/LocalRouter';
+import useCurrentPointName from 'lib/useCurrentPointName';
+import * as need from 'lib/need';
+
+import { usePointCursor } from 'store/pointCursor';
+import { usePointCache } from 'store/pointCache';
+
+const NAMES = {
+  REQUESTS: 'REQUESTS',
+  ALL: 'ALL',
+};
+
+const OPTIONS = [
+  { text: 'Requests', value: NAMES.REQUESTS },
+  { text: 'All', value: NAMES.ALL },
+];
+
+const VIEWS = {
+  [NAMES.REQUESTS]: Tab,
+  [NAMES.ALL]: Tab,
+};
+
+function Tab({ className, points, onAccept, onDecline }) {
+  if (Nothing.hasInstance(points)) {
+    return <Grid className={className}></Grid>;
+  }
+
+  const _points = points.getOrElse([]);
+
+  return (
+    <Grid className={className}>
+      {_points.length === 0 && (
+        <Grid.Item full as={HelpText} className="mt8 t-center">
+          {Just.hasInstance(points) ? (
+            'No points to display'
+          ) : (
+            <>
+              <Blinky /> Loading...
+            </>
+          )}
+        </Grid.Item>
+      )}
+
+      {_points.map(point => (
+        <Resident
```

Comment:
This list can get huge! Are we fine rendering all those in one go? Even if performance is fine, that's one big list to navigate. Maybe want pagination? cc @jyng

Reply:
only 1024 items for now ;) but yeah i agree should be paginated in some way